src/Interpreters/Streaming/Aggregator.cpp

```diff
@@ -1468,41 +1458,36 @@ BlocksList Aggregator::mergeAndConvertTwoLevelToBlocksImpl(
         return blocks;
     };
 
-    /// packaged_task is used to ensure that exceptions are automatically thrown into the main stream.
-    std::vector<std::packaged_task<BlocksList()>> tasks(num_threads);
+    auto num_threads = std::min(max_threads, buckets.size());
+    if (num_threads <= 1)
+        converter(0, nullptr);
+
+    /// Process in parallel
+    /// proton FIXME : separate final vs non-final converting. For non-final converting, we don't need
+    /// each arena for each thread.
+    for (size_t i = first.aggregates_pools.size(); i < num_threads; ++i)
+        first.aggregates_pools.push_back(std::make_shared<Arena>());
+
+    auto results = std::make_shared<std::vector<BlocksList>>();
+    results->resize(num_threads);
+    ThreadPool thread_pool(num_threads);
```

Comment:
nope, for multiple threads, when we run the `N`st task, which may throw an error. So we need cancel other tasks thread to avoid useless merging and converting

Reply:
1) it is pure computing, execetption shall be rare
2) A SCOPE_EXIT to cancel the threads is more elegant than try catch it